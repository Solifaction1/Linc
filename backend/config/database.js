const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
