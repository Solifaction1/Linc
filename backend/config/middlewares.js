module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
