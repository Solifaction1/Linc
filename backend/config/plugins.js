module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
