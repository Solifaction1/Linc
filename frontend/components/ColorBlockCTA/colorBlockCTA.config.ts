import ColorBlockData from '@models/colorBlockCTA.model'

const defaultProps: ColorBlockData = {
	ctas: [
		{
			description: {
				text: '<p>description bottom left</p>',
				className: 'string',
			},

			cta: {
				id: 0,
				title: 'Afspraak maken',
				link: 'https://google.be/left',
				externalLink: false,
				buttonType: 'Button primary',
			},
		},
		{
			description: {
				text: '<p>description bottom right</p>',
				className: 'string',
			},

			cta: {
				id: 0,
				title: 'Contact',
				link: 'https://google.be/right',
				externalLink: false,
				buttonType: 'Button primary',
			},
		},
	],
}

export { defaultProps }
