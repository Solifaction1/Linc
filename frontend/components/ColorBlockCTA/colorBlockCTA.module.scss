@import 'styles/vars';
@import 'styles/utilities';

.colorBlockCTA {
	@extend %page-container;
	position: relative;
	z-index: 3;
	grid-template-areas: 'left left left left left left right right right right right right';
	grid-gap: 0;

	width: 100%;
	margin-top: calc(#{$spacing-base} * 0);
	margin-right: 0;
	margin-left: 0;
	padding-right: 0;
	padding-bottom: 0;
	padding-left: 0;

	@media (max-width: $break-lg) {
		grid-template-areas:
			'left left left left left left left left left left left left'
			'right right right right right right right right right right right right';
	}

	.leftColorBlock {
		grid-area: left;
		background-color: $color-blue;
		color: $color-blue-dark;

		.leftContainer {
			margin-left: auto;
			padding: calc(#{$spacing-base} * 8) calc(#{$spacing-base} * 12)
				calc(#{$spacing-base} * 8) 0;

			@media (max-width: $break-xlg) {
				padding: calc(#{$spacing-base} * 8) calc(#{$spacing-base} * 12)
					calc(#{$spacing-base} * 8) 3.6rem;
			}

			.text {
				p,
				h2,
				h1,
				ul,
				ol,
				li {
					color: $color-blue-dark;
				}
			}
		}
	}

	.rightColorBlock {
		grid-area: right;
		background-color: $color-blue-dark;
		color: $color-blue;

		.rightContainer {
			margin-right: auto;
			padding: calc(#{$spacing-base} * 8) 0 calc(#{$spacing-base} * 8)
				calc(#{$spacing-base} * 11);

			@media (max-width: $break-xlg) {
				padding: calc(#{$spacing-base} * 8) 3.6rem calc(#{$spacing-base} * 8)
					calc(#{$spacing-base} * 11);
			}

			.text {
				p,
				h2,
				h1,
				ul,
				ol,
				li {
					color: $color-blue;
				}
			}
		}
	}

	.leftColorBlock,
	.rightColorBlock {
		display: flex;
		align-items: center;
		min-height: 46.6rem;

		.leftContainer,
		.rightContainer {
			display: flex;
			flex-direction: column;
			justify-content: space-between;

			max-width: 57.15rem;
			height: 100%;

			.cta {
				margin-top: 4rem;
			}

			@media (max-width: $break-lg) {
				justify-content: center;

				max-width: unset;
				margin-right: unset;
				margin-left: unset;
				padding: 7.5rem 3.6rem;
			}

			@media (max-width: $break-s) {
				padding: 7.5rem 2.4rem;
			}
		}
	}
}
