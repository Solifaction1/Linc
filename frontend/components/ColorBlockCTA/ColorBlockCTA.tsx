import React from 'react'
import ColorBlockData from '@models/colorBlockCTA.model'
import YGWYS from '@components/YGWYS/YGWYS'
import SwitchCTA from '@templates/SwitchCTA'
import { defaultProps } from './colorBlockCTA.config'

import styles from './colorBlockCTA.module.scss'

const ColorBlockCTA = (props: ColorBlockData) => {
	return (
		<article className={styles.colorBlockCTA}>
			<div className={styles.leftColorBlock}>
				<div className={styles.leftContainer}>
					{props.ctas[0].description?.text && (
						<div className={styles.text}>
							<YGWYS text={props.ctas[0].description?.text} />
						</div>
					)}
					{props.ctas[0].cta?.link && (
						<div className={styles.cta}>
							<SwitchCTA ctaData={props.ctas[0].cta} color='lightBlue'/>
						</div>
					)}
				</div>
			</div>

			<div className={styles.rightColorBlock}>
				<div className={styles.rightContainer}>
					{props.ctas[1].description?.text && (
						<div className={styles.text}>
							<YGWYS text={props.ctas[1].description?.text} />
						</div>
					)}
					{props.ctas[1].cta?.link && (
						<div className={styles.cta}>
							<SwitchCTA ctaData={props.ctas[1].cta} color='darkBlue'/>
						</div>
					)}
				</div>
			</div>
		</article>
	)
}

ColorBlockCTA.defaultProps = defaultProps

export default ColorBlockCTA
