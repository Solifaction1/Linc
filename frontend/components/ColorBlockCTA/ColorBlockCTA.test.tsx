import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { defaultProps } from './colorBlockCTA.config'
import ColorBlockCTA from './ColorBlockCTA'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('ColorBlockCTA', () => {
	it('renders an ColorBlockCTA', () => {
		const { container, getByText } = render(<ColorBlockCTA {...defaultProps} />)
		expect(container).toMatchSnapshot()

		expect(container.querySelector('article')).toBeVisible()
		expect(container.getElementsByClassName('colorBlockCTA')[0]).toBeVisible()

		expect(container.getElementsByClassName('leftColorBlock')[0]).toBeVisible()
		expect(container.getElementsByClassName('leftContainer')[0]).toBeVisible()
		expect(container.getElementsByClassName('text')[0]).toBeVisible()
		expect(container.getElementsByClassName('ygwys')[0]).toBeVisible()
		expect(
			getByText('description bottom left', { selector: 'p' })
		).toBeVisible()
		expect(container.getElementsByClassName('cta')[0]).toBeVisible()
		const btnLeft = container.querySelectorAll('button')[0]
		expect(btnLeft?.getAttribute('class')).toEqual(
			'button false primary md false  lightBlue'
		)
		expect(btnLeft?.getAttribute('type')).toEqual('button')
		expect(getByText('Afspraak maken', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('rightColorBlock')[0]).toBeVisible()
		expect(container.getElementsByClassName('rightContainer')[0]).toBeVisible()
		expect(container.getElementsByClassName('text')[1]).toBeVisible()
		expect(container.getElementsByClassName('ygwys')[1]).toBeVisible()
		expect(
			getByText('description bottom right', { selector: 'p' })
		).toBeVisible()
		expect(container.getElementsByClassName('cta')[1]).toBeVisible()
		const btnRight = container.querySelectorAll('button')[1]
		expect(btnRight?.getAttribute('class')).toContain(
			'button false primary md false  darkBlue'
		)
		expect(btnRight?.getAttribute('type')).toEqual('button')
		expect(getByText('Contact', { selector: 'p' })).toBeVisible()
	})
})
