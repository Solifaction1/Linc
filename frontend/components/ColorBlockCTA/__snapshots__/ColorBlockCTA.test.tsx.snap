// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBlockCTA renders an ColorBlockCTA 1`] = `
<div>
  <article
    class="colorBlockCTA"
  >
    <div
      class="leftColorBlock"
    >
      <div
        class="leftContainer"
      >
        <div
          class="text"
        >
          <div
            class="ygwys 
			undefined undefined"
          >
            <p>
              description bottom left
            </p>
          </div>
        </div>
        <div
          class="cta"
        >
          <button
            class="button false primary md false  lightBlue"
            type="button"
          >
            <p
              class="label"
            >
              Afspraak maken
            </p>
          </button>
        </div>
      </div>
    </div>
    <div
      class="rightColorBlock"
    >
      <div
        class="rightContainer"
      >
        <div
          class="text"
        >
          <div
            class="ygwys 
			undefined undefined"
          >
            <p>
              description bottom right
            </p>
          </div>
        </div>
        <div
          class="cta"
        >
          <button
            class="button false primary md false  darkBlue"
            type="button"
          >
            <p
              class="label"
            >
              Contact
            </p>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;
